src/js/slide/get-thumb-bounds.js

```diff
@@ -80,29 +82,32 @@ export function getThumbBounds(index, itemData, instance) {
   }
 
   const { element } = itemData;
+  /** @type {Bounds | undefined} */
   let thumbBounds;
-  /** @type {HTMLElement} */
-  let thumbnail;
+  /** @type {HTMLElement | null} */
+  let thumbnail = null;
 
   if (element && instance.options.thumbSelector !== false) {
     const thumbSelector = instance.options.thumbSelector || 'img';
     thumbnail = element.matches(thumbSelector)
       ? element : element.querySelector(thumbSelector);
   }
 
-  thumbnail = instance.applyFilters('thumbEl', thumbnail, itemData, index);
-
   if (thumbnail) {
+    thumbnail = instance.applyFilters('thumbEl', thumbnail, itemData, index);
```

Comment:
Allow filtering even if the thumbnail is missing initially

Reply:
fixed and updated types for filters